### PR TITLE
os/bluestore/KernelDevice: misc cleanup

### PR DESCRIPTION
**Pull request by majianpeng**

### CODE REVIEW

File: src/os/bluestore/KernelDevice.cc [hunk 1]
```diff
@@ -94,6 +94,12 @@ int KernelDevice::open(const string& p)
     derr << __func__ << " open got: " << cpp_strerror(r) << dendl;
     goto out_fail;
   }
+  r = posix_fadvise(fd_direct, 0, 0, POSIX_FADV_RANDOM);
```
Comment (majianpeng) [inline src/os/bluestore/KernelDevice.cc]: @ukernel . Have you time to review this commit? Thanks!

**Comment by majianpeng** [comment]
@liewegas .ping.

**Comment by majianpeng** [comment]
@liewegas . ping! Thanks.

File: src/os/bluestore/KernelDevice.cc (hunk 1 above)
Comment (liewegas) [reply src/os/bluestore/KernelDevice.cc]: I think this isn't strictly necessary because direct io won't trigger readahead, but it doesn't hurt!

**Comment by liewegas** [comment]
retest this please

**Comment by majianpeng** [comment]
update: remove the patch which set FDAV_RANDOM for direct fd;  rebase on master.

**Comment by liewegas** [comment]
http://pulpito.ceph.com/sage-2018-05-23_14:50:29-rados-wip-sage2-testing-2018-05-22-1410-distro-basic-smithi/

**Pull request merged**